code-challenges.test.js

```diff
@@ -13,47 +16,151 @@
 // --------------------INSTRUCTOR EXAMPLE: Create a function that takes in an array of numbers and returns an array with all the numbers multiplied by 3.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const numbersArray1 = [6, 7, 8, 9, 10]
-// Expected output: [18, 21, 24, 27, 30]
-const numbersArray2 = [24, 27, 30, 33, 36]
-// Expected output: [72, 81, 90, 99, 108]
-
+// I think the proper format for the test would look something like line 25 thru 29 and the function i created is on line 30 thru 32. yet when i run this test in jest, i get a failed result.. but not the undefined one that we expect. not sure why.
+
+const numbersArray1 = [6, 7, 8, 9, 10];
+// // Expected output: [18, 21, 24, 27, 30]
+const numbersArray2 = [24, 27, 30, 33, 36];
+// // Expected output: [72, 81, 90, 99, 108]
+describe("multThree", () => {
+  it("returns a new array with the numbers multiplied by three", () => {
+    expect(multThree(numbersArray1)).toEqual(18, 21, 24, 27, 30);
+    expect(multThree(numbersArray2)).toEqual(72, 81, 90, 99, 108);
+  });
+});
+const multThree = (array) => {
+  return array() * 3;
+};
 // b) Create the function that makes the test pass.
 
 // --------------------1) Create a function that takes a object as an argument and decides if the number inside it is evenly divisible by three or not.
+//input= an object as an argument
+//output= wether the number is evenly divisible by 3
+// i think a conditional statement with the funtion to return wether the number is evenly divisible by three or not is the appropriate approach
 
 // a) Create a test with expect statements for each of the variables provided.
+//input= number
+//output= is divisible by three || is not divisible by three
 
-const object1 = { number: 15 }
+const object1 = { number: 15 };
 // Expected output: "15 is divisible by three"
-const object2 = { number: 0 }
+const object2 = { number: 0 };
 // Expected output: "0 is divisible by three"
-const object3 = { number: -7 }
+const object3 = { number: -7 };
 // Expected output: "-7 is not divisible by three"
+describe("divByThree", () => {
+  it("returns wether the number is divisible by three or not based on input", () => {
+    expect(divByThree(object1)).toEqual("15 is divisible by three");
+    expect(divByThree(object2)).toEqual("0 is divisible by three");
+    expect(divByThree(object3)).toEqual("-7 is not divisible by three");
+  });
+});
 
 // b) Create the function that makes the test pass.
+// i want to make a function inside of a conditional statement i can use for any number. i think it is a called a dynamic code. i will name it divByThree and make the if statement to return if is divisible by three or else return the statement that it is not divisible by three.
```

Comment:
This code is definitely dynamic and super close to being exactly what I am looking for.  Since you are passing an object into the function, you would also need to access the value from the key number.  Ex: if object was the param, you can access the value of number by using dot notation: object.number